zorg/buildbot/builders/ClangBuilder.py

```diff
@@ -427,9 +436,31 @@ def _getClangCMakeBuildFactory(
         # Note: Backslash path separators do not work well with cmake and ninja.
         # Forward slash path separator works on Windows as well.
         stage1_cc = InterpolateToPosixPath(
-                        f"-DCMAKE_C_COMPILER=%(prop:builddir)s/{stage1_install}/bin/{cc}")
+                        f"%(prop:builddir)s/{stage1_install}/bin/{cc}")
         stage1_cxx = InterpolateToPosixPath(
-                        f"-DCMAKE_CXX_COMPILER=%(prop:builddir)s/{stage1_install}/bin/{cxx}")
+                        f"%(prop:builddir)s/{stage1_install}/bin/{cxx}")
+
+        # If stage2_toolchain_options is set when we'll use a toolchain file
+        # to specify the compiler being used (the just-built stage1) and add
+        # any stage2_toolchain_options to it. Otherwise, just set
+        # -DCMAKE_{C,CXX}_COMPILER.
+        if stage2_toolchain_options is None:
+            compiler_args = [
+                    f"-DCMAKE_C_COMPILER={stage1_cc}",
+                    f"-DCMAKE_CXX_COMPILER={stage1_cxx}"
+            ]
+        else:
+            toolchain_file = InterpolateToPosixPath(
+                        f"%(prop:builddir)s/{stage2_build}/stage1-toolchain.cmake")
+            with open(toolchain_file, 'w') as file:
+                file.write(f"set(CMAKE_C_COMPILER {stage1_cc})\n")
+                file.write(f"set(CMAKE_CXX_COMPILER {stage1_cxx})\n")
+                for option in stage2_toolchain_options:
+                    file.write(f"{option}\n")
```

Comment:
I'm not sure whether this runs on the build master or the worker. If its the former, it might need to be a step "write toolchain file" that echos the contents into a file on the worker.

Reply:
Good catch, I think it's likely this isn't writing to a worker-local file as I hope.

Reply:
I've fixed this now.